integtest/src/main/scala/com/pingcap/spark/TestCase.scala

```diff
@@ -57,11 +57,17 @@ class TestCase(val prop: Properties) extends LazyLogging {
 
   private val eps = 1.0e-2
 
-  protected var testsFailed = 0
-  protected var testsExecuted = 0
-  protected var testsSkipped = 0
-  protected var inlineSQLNumber = 0
-  protected var ignoredTest = 0
+  @volatile protected var testsFailed = 0
```

Comment:
Atomic operation like a += 1 cannot be protected by volatile only.
a += 1 -> a = a + 1 involve both load and store 2 operations. When load, add and store, volatile does not ensure the original value you load is not changed by other threads.
Use AtomicInteger instead.